test/test_SUITE.erl

```diff
@@ -0,0 +1,128 @@
+
+    rabbit_ct_helpers:run_setup_steps(Config1,
+                                      [ fun configure_backend/1 ] ++
+                                      rabbit_ct_broker_helpers:setup_steps()).
+
+end_per_suite(Config) ->
+    rabbit_ct_helpers:run_teardown_steps(Config,
+                                         rabbit_ct_broker_helpers:teardown_steps()).
+
+init_per_group(_, Config) -> Config.
+
+end_per_group(_, Config) -> Config.
+
+init_per_testcase(with_backend = Testcase, Config) ->
+    PrivDir = ?config(priv_dir, Config),
+    DataDir = ?config(data_dir, Config),
+    BuildDir = filename:join([PrivDir, "build"]),
+    ok = filelib:ensure_dir(filename:join(BuildDir, "dummy")),
+    AmqpPort = rabbit_ct_broker_helpers:get_node_config(Config, 0, tcp_port_amqp),
+    Script = filename:join([DataDir, "run_backend.sh"]),
+    Port = erlang:open_port({spawn, Script}, [use_stdio,
+                                              stderr_to_stdout,
+                                              exit_status,
+                                              {env, [{"BUILD", BuildDir},
+                                                     {"AMQP_PORT", integer_to_list(AmqpPort)}
+                                                    ]}]),
+    Config1 = rabbit_ct_helpers:set_config(Config, {backend_port, Port}),
+    % wait for backend to start
+    timer:sleep(5000),
+    rabbit_ct_helpers:testcase_started(Config1, Testcase).
+
+end_per_testcase(with_backend = Testcase, Config) ->
+    Port = ?config(backend_port, Config),
+    print_port_data(Port, []),
+    {os_pid, OsPid} = erlang:port_info(Port, os_pid),
+    ct:pal("backend os pid ~p", [OsPid]),
+    KillCmd =
+        case os:type() of
+            {unix, _} -> io_lib:format("kill ~p", [OsPid]);
+            {win32, _} -> io_lib:format("taskkill /PID ~p", [OsPid])
+        end,
+    os:cmd(KillCmd),
+    true = erlang:port_close(Port),
+    rabbit_ct_helpers:testcase_finished(Config, Testcase).
+
+print_port_data(Port, Acc) ->
+    receive
+        {Port, {exit_status, X}} ->
+            ct:pal("port exited with ~p", [X]),
+            ct:pal("backend: ~p", [Acc]);
+        {Port, {data, Out}} ->
+            print_port_data(Port, Acc ++ [Out])
+    after 5000 ->
+            ct:pal("backend: ~p", [Acc])
+    end.
+
+%% -------------------------------------------------------------------
+%% Testcases.
+%% -------------------------------------------------------------------
+
+with_backend(Config) ->
+
+    AmqpPort = rabbit_ct_broker_helpers:get_node_config(Config, 0, tcp_port_amqp),
+    Host = rabbit_ct_helpers:get_config(Config, rmq_hostname),
+    {ok, Con} = amqp_connection:start(#amqp_params_network{host = Host,
+                                                           port = AmqpPort,
+                                                           username = <<"simon">>,
+                                                           password = <<"simon">>}),
```

Comment:
ok I could repro cause I had some java processes still running that hadn't been properly killed from previous runs. Perhaps SIGTERM isn't enough after all?